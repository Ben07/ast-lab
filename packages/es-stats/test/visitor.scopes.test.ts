import fs from 'fs';
import path from 'path';
import { parse } from '@babel/parser';
import traverse from '@babel/traverse';
import debug from 'debug';
import { MemberRelation } from 'ast-lab-types';
import createVisitor from '../src/visitors/rootRelation';

const dirs = ['scopes', 'imports', 'exports'];
describe('rootRelation visitors', () => {
  dirs.forEach(dir => {
    const dirPath = path.resolve(__dirname, '__fixtures__', dir);
    const files = fs.readdirSync(dirPath);
    files.forEach(file => {
      test(`${dir}/${file}`, () => {
        debug('es-stats:test')(`${dir}/${file}`);
        let res = {} as MemberRelation;
        const code = fs.readFileSync(path.resolve(dirPath, file), 'utf-8');
        traverse(
          // @ts-ignore
          parse(code, {
            sourceType: 'module',
            plugins: ['classProperties', 'dynamicImport'],
          }),
          createVisitor(res)
        );
        expect(res).toMatchSnapshot();
      });
    });
  });
});
