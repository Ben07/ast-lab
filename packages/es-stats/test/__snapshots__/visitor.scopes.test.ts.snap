// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootRelation visitors exports/exportAllFrom.js 1`] = `Object {}`;

exports[`rootRelation visitors exports/exportDefault.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors exports/exportDefaultFuncAlias.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors exports/exportDefaultValueAlias.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors exports/exportFunc.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors exports/exportFuncAlias.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors exports/exportNamed.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors exports/exportNamedFrom.js 1`] = `
Object {
  "a": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors exports/noExports.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors imports/dynamicImport-asyncawait.js 1`] = `
Object {
  "func": Array [],
}
`;

exports[`rootRelation visitors imports/dynamicImport-then.js 1`] = `
Object {
  "func": Array [],
}
`;

exports[`rootRelation visitors imports/importAlias.js 1`] = `
Object {
  "c": Array [
    "b",
  ],
}
`;

exports[`rootRelation visitors imports/importAll.js 1`] = `Object {}`;

exports[`rootRelation visitors imports/importCss.js 1`] = `Object {}`;

exports[`rootRelation visitors imports/importDefault.js 1`] = `Object {}`;

exports[`rootRelation visitors imports/importNamed.js 1`] = `Object {}`;

exports[`rootRelation visitors imports/noImports.js 1`] = `
Object {
  "a": Array [],
}
`;

exports[`rootRelation visitors scopes/arrowfunction.js 1`] = `
Object {
  "DEFAULT": Array [],
  "a": Array [],
  "arrow": Array [
    "a",
    "DEFAULT",
    "b",
  ],
  "b": Array [],
}
`;

exports[`rootRelation visitors scopes/asyncFunction.js 1`] = `
Object {
  "a": Array [],
  "asyncFunc": Array [
    "a",
  ],
  "b": Array [],
}
`;

exports[`rootRelation visitors scopes/class.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "c": Array [
    "b",
  ],
}
`;

exports[`rootRelation visitors scopes/constant.js 1`] = `
Object {
  "a": Array [],
  "exportArray": Array [
    "a",
  ],
  "exportArrayMember": Array [
    "exportArray",
  ],
  "exportConstant": Array [
    "a",
  ],
  "exportObject": Array [
    "a",
  ],
  "exportObject2": Array [
    "a",
  ],
  "exportObjectMember": Array [
    "exportObject",
  ],
  "exportRestArray": Array [
    "exportArray",
  ],
  "exportRestObject": Array [
    "exportObject",
  ],
}
`;

exports[`rootRelation visitors scopes/eval.js 1`] = `
Object {
  "a": Array [],
  "func": Array [
    "eval",
  ],
}
`;

exports[`rootRelation visitors scopes/forloop.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "func": Array [
    "b",
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/function.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "func": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/function-argsObjectAssign.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "func": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/function-argsSpread.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "func": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/generator.js 1`] = `
Object {
  "a": Array [],
  "func": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/ifelse.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "func": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/importExport.js 1`] = `
Object {
  "default": Array [
    "im",
  ],
}
`;

exports[`rootRelation visitors scopes/switch.js 1`] = `
Object {
  "TYPES": Array [],
  "func": Array [
    "type",
    "TYPES",
    "v1",
  ],
  "type": Array [],
  "v1": Array [],
  "v2": Array [],
}
`;

exports[`rootRelation visitors scopes/template.js 1`] = `
Object {
  "a": Array [],
  "func": Array [
    "a",
  ],
}
`;

exports[`rootRelation visitors scopes/trycatch.js 1`] = `
Object {
  "a": Array [],
  "b": Array [],
  "func": Array [
    "console",
    "a",
  ],
}
`;
